xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/adapter/remote/service/impl/RemoteAccountInformationServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package de.adorsys.xs2a.adapter.remote.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.adorsys.xs2a.adapter.mapper.TransactionsReportMapper;
+import de.adorsys.xs2a.adapter.model.BookingStatusTO;
+import de.adorsys.xs2a.adapter.model.TransactionsResponse200JsonTO;
+import de.adorsys.xs2a.adapter.remote.api.AccountInformationClient;
+import de.adorsys.xs2a.adapter.service.RequestHeaders;
+import de.adorsys.xs2a.adapter.service.RequestParams;
+import de.adorsys.xs2a.adapter.service.Response;
+import de.adorsys.xs2a.adapter.service.model.AccountReference;
+import de.adorsys.xs2a.adapter.service.model.TransactionsReport;
+import org.junit.Test;
+import org.mapstruct.factory.Mappers;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class RemoteAccountInformationServiceTest {
+
+    @Test
+    public void getTransactionListAsString() {
+        AccountInformationClient client = mock(AccountInformationClient.class);
+        RemoteAccountInformationService service = new RemoteAccountInformationService(client);
+
+        when(client.getTransactionListAsString(any(), any(), any(), anyString(), any(), anyBoolean(), anyBoolean(), anyMap())).thenReturn(buildResponseEntity());
+
+        Response<String> response = service.getTransactionListAsString(
+            "accountId",
+            RequestHeaders.fromMap(buildHeaders()),
+            RequestParams.fromMap(buildRequestParams())
+        );
+
+        assertThat("{\"bookingStatus\":\"booked\"}", is(response.getBody()));
```

Comment:
booking status is not a transaction report field, how is this working?

Reply:
thanks, updated